doc/release-notes-1.2.0.rst

```diff
@@ -0,0 +1,490 @@
+    1 indicates that further processing is required by the :cpp:func:`mqtt_evt_handler` that called :cpp:func:`aws_fota_mqtt_evt_handler`.
+
+* :ref:`lib_nrf_cloud`:
+
+  * Removed the button/switch pairing method from nRF Cloud FSM. ??? what is FSM?
+  * Added functionality to handle the device configuration in the device shadow.
+
+* :ref:`liblwm2m_carrier_readme`:
+
+  * Updated to version 0.8.1.
+
+* Moved the following libraries from ``drivers/`` to ``lib/``:
+
+  * :ref:`at_cmd_readme`
+  * ``lte_link_control``
+
+
+Updated drivers
+---------------
+
+* Moved the following drivers from ``drivers/`` to ``drivers/gps/``:
+
+  * ``gps_sim``
+  * ``nrf9160_gps``
+
+
+BSD library
+-----------
+
+* Updated the :ref:`nrfxlib:bsdlib` to version 0.6.1.
+
+
+Common libraries, drivers, and samples
+======================================
+
+* Added the following libraries:
+
+  * ??? fprotect?
+
+Updated samples and applications
+--------------------------------
+
+* :ref:`bootloader`:
+
+  * Moved the provisioning data (slot sizes/addresses and public keys) to one-time programmable (OTP) memory.
+  * Implemented invalidation of public keys.
+
+
+Updated libraries
+-----------------
+
+* :ref:`doc_fw_info`:
+
+  * Renamed ABIs to EXT_APIs.
+  * Restructured the :c:type:`fw_info` structure:
+
+    * Renamed the fields ``firmware_size``, ``firmware_address``, and ``firmware_version`` to ``size``, ``address``, and ``version``.
+    * Added a field to invalidate the structure.
+    * Added reserved fields for future use.
+    * EXT_APIs are now in a list at the end of the structure, instead of being available behind a function call.
+    * EXT_APIs can now be requested by adding a request structure to a list after the EXT_API list itself.
+
+  * Updated how EXT_API requests are checked.
+    Requests are now checked against EXT_APIs by the bootloader before booting the image.
+  * Added two new allowed offsets for the struct: 0x0 and 0x1000 bytes.
+  * Removed ``memeq()`` in favor of regular ``memcmp()``.
+  * Renamed ``__ext_api()`` to ``EXT_API()``, because names starting with ``__`` are reserved for the compiler.
+  * Added new configuration options ``CONFIG_*_EXT_API_REQUIRED`` and ``CONFIG_*_EXT_API_ENABLED`` for, respectively, users and providers of EXT_APIs.
+
+* :ref:`lib_dfu_target`:
+
+  * Added the configuration option :option:`CONFIG_DFU_TARGET_MCUBOOT_SAVE_PROGRESS`, which uses Zephyr's :ref:`zephyr:settings` subsystem.
+    When this option is enabled, the write progress of an MCUboot style upgrade is stored, so that the progress is retained when the device reboots.
+  * Fixed a bug where :cpp:func:`dfu_target_done` printed ``unable to deinitialize dfu resource`` when no target was initialized. ??? did it actually print it or return an error?
+
+* Moved the following libraries from ``drivers/`` to ``lib/``:
+
+  * :ref:`fprotect_readme`
+  * :ref:`st25r3911b_nfc_readme`
+  * ``adp536x``
+  * ``flash_patch``
+
+
+Crypto
+======
+
+???
```

Comment:
Crypto additions

* Added low-level cryptographic test suite using NIST, RFCs and custom test vectors.
* Added nrf_cc310_mbedcrypto/nrf_cc310_platform library version 0.9.2:
  * Fixes power-efficiency issues. 
  * Library adds experimental use of CryptoCell interrupt instead of busy-waits.

hw_cc310 driver addition
* Added support for CryptoCell interrupt in driver.